app/src/main/java/space/taran/arkmemo/data/viewmodels/EditTextNotesViewModel.kt

```diff
@@ -1,25 +1,41 @@
 package space.taran.arkmemo.data.viewmodels
 
-import android.content.Context
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
-import space.taran.arkmemo.data.repositories.TextNotesRepository
+import kotlinx.coroutines.withContext
+import space.taran.arkmemo.data.repositories.TextNotesRepo
 import space.taran.arkmemo.models.TextNote
+import space.taran.arkmemo.preferences.MemoPreferences
 import javax.inject.Inject
 
 @HiltViewModel
```

Comment:
Please do me a favor. If you have time, it would be nice if you can move all view models to `ui` package. ViewModel is a part of the presentation layer and it controls the state to be displayed in the screen. It does not belong to data layer